### DISCUSSION TITLE
How can I control the width of DataGrid columns?

### DISCUSSION THREAD

**Discussion by Gigas002** [AvaloniaUI/Avalonia#7612] [Q&A] [answered]
Here's the DataGrid I have:

![image](https://user-images.githubusercontent.com/24297712/154098232-e60b8fc7-2726-4960-bf01-7298c0a3ce0a.png)

It's created this way on the `View`:

```xml
<DataGrid AutoGenerateColumns="True" IsReadOnly="True" Items="{Binding Dependencies}"/>
```

It seems, like the default value for `DataGrid`'s column's width is `Auto`. How can I force it to use `*` for, for example, the `Name` columns here? And how can I change this `DataGrid`'s header's alignment (e.g. `center` for all headers)?

**Comment by luthfiampas** [comment, marked as answer]
You can use `ColumnWidth` property. But if you want to control the width of individual column, you will need to define all the columns manually.

```xaml
<DataGrid AutoGenerateColumns="False" Items="{Binding Users}">
  <DataGrid.Columns>
    <DataGridTextColumn Header="Name" Width="*" Binding="{Binding Name}" />
    <DataGridTextColumn Header="Email" Width="3*" Binding="{Binding Email}" />
  </DataGrid.Columns>
</DataGrid>
```

To align column header text to center, add this to your style.
```xaml
<Style Selector="DataGridColumnHeader">
  <Setter Property="HorizontalContentAlignment" Value="Center" />
</Style>
```

**Reply by Gigas002** [reply]
Thanks for solution, it works as expected. Could you please also explain, where to add that style part of code? I don't use a separate file with defined styles, can I define these in my `View.axaml`?

**Reply by Gigas002** [reply]
Ah, don't mind, I've found out the way
